prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/metrics/impl/MetricsManager.java

```diff
@@ -0,0 +1,332 @@
+        }
+
+        // Restart the callback reporter
+        if (callbackReporter != null) {
+            callbackReporter.close();
+            callbackReporter = null;
+        }
+
+        if (callbackReporter == null) {
+            callbackReporter = new CallbackReporter(this::reporterCallback, registry, metricsProvider);
+            callbackReporter.start(updateRate, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void disableMetrics() {
+        isActive = false;
+
+        if (consoleReporter != null) {
+            consoleReporter.close();
+            consoleReporter = null;
+        }
+
+        if (callbackReporter != null) {
+            callbackReporter.close();
+            callbackReporter = null;
+        }
+
+        // Clear metrics
+        clear();
+    }
+
+    @Override
+    public void clear() {
+        // Remove all metrics but the gauges
+        final Set<String> toRemove = new HashSet<>(registry.getNames());
+        toRemove.removeAll(registry.getGauges().keySet());
+        registry.removeMatching((name, metric) -> toRemove.contains(name));
+    }
+
+    @Override
+    public void enableMetrics(String... names) {
+        if (names != null) {
+            activeMetrics.addAll(Arrays.asList(names));
+        }
+    }
+
+    @Override
+    public void disableMetrics(String... names) {
+        if (names != null) {
+            activeMetrics.removeAll(Arrays.asList(names));
+        }
+    }
+
+    @Override
+    public IMetricTimer withTimer(String name) {
+        if (!isEnabled(name)) {
+            return new DummyTimer(name);
+        }
+
+        return new MetricsTimer(registry, name);
+    }
+
+    @Override
+    public <T> void registerGauge(final String name, final Callable<T> gaugeCallback) {
+        if (registry != null && gaugeCallback != null) {
+            registry.registerGauge(name, new Gauge<T>() {
+                @Override
+                public T getValue() {
+                    try {
+                        return gaugeCallback.call();
+                    } catch (Exception e) {
+                        logger.error("Error calling gauge {}: {}", name, e.getMessage(), e);
+                        return null;
+                    }
+                }
+            });
+        }
+    }
+
+    @Override
+    public void unregisterGauge(String name) {
+        if (registry != null) {
+            registry.remove(name);
+        }
+    }
+
+    @Override
+    public void unregisterGaugesByPrefix(String prefix) {
+        if (registry != null && prefix != null && !prefix.isBlank()) {
+            for (String key : registry.getGauges().keySet()) {
+                if (key.startsWith(prefix)) {
+                    registry.remove(key);
+                }
+            }
+        }
+    }
+
+    @Override
+    public IMetricCounter getCounter(String name) {
+        if (!isEnabled(name)) {
+            // Return a dummy counter
+            return new DummyCounter(name);
```

Comment:
What happens if the configuration is updated to enable it after the counter is obtained? Shouldn't the Counter implementation stay the same but just ignore the calls when it is disabled?